### PR TITLE
Use prtconf to determine CPU type on AIX

### PR DESCRIPTION
**Pull request by martin-frbg**
for #1803

### CODE REVIEW

**Comment by ayappanec** [comment]
Power 9 released recently. We can include that as well.

**Comment by martin-frbg** [comment]
Good point, thanks. Mind you I have no way to test this.

File: cpuid_power.c [hunk 1]
```diff
@@ -120,13 +124,41 @@ int detect(void){
...
+  pclose(infile);
+
+  if (!strncasecmp(p, "POWER3", 6)) return CPUTYPE_POWER3;
+  if (!strncasecmp(p, "POWER4", 6)) return CPUTYPE_POWER4;
+  if (!strncasecmp(p, "PPC970", 6)) return CPUTYPE_PPC970;
+  if (!strncasecmp(p, "POWER5", 6)) return CPUTYPE_POWER5;
+  if (!strncasecmp(p, "POWER6", 6)) return CPUTYPE_POWER6;
+  if (!strncasecmp(p, "POWER7", 6)) return CPUTYPE_POWER6;
+  if (!strncasecmp(p, "POWER8", 6)) return CPUTYPE_POWER8;
+  if (!strncasecmp(p, "POWER8", 6)) return CPUTYPE_POWER8;
```
Comment (sbutcher) [inline cpuid_power.c]: should this say 
`   if (!strncasecmp(p, "POWER9", 6)) return CPUTYPE_POWER8; `
instead?

**Review by sbutcher — COMMENTED** [review]
I'm not familiar with the code, but we have a POWER9 and it built successfully with these changes, _however_ i noticed a couple of things i'd like to query. 
Also, not in this PR, but perhaps should be: is line 183 also wrong by having 2 `returns`?
`return return CPUTYPE_POWER8;`

File: cpuid_power.c (hunk 1 above)
Comment (martin-frbg) [reply cpuid_power.c]: Err, yes... and I guess the double "return" in 183 proves that I was never good at copypasting.

**Comment by sbutcher** [comment]
Thanks. Can confirm this builds for me on POWER9, and I was able to compile `test_cblas_dgemm.c` against this build.

**Comment by martin-frbg** [comment]
Thanks - and just to confirm, your resulting library has "power9" in its name now rather than the earlier "power5" (and the build used the POWER8 target) ?

**Comment by sbutcher** [comment]
We built using easybuild, and it produces
```
libopenblas_power8p-r0.3.4.dev.a
libopenblas_power8p-r0.3.4.dev.so
```
(no longer requires manually adding POWER8 target by hand)

Checkout from git and running `make` without options produces
```penBLAS build complete. (BLAS CBLAS LAPACK LAPACKE)

  OS               ... Linux             
  Architecture     ... power               
  BINARY           ... 64bit                 
  C compiler       ... GCC  (command line : /usr/bin/gcc)
  Fortran compiler ... GFORTRAN  (command line : /usr/bin/gfortran)
  Library Name     ... libopenblas_power8p-r0.3.4.dev.a (Multi threaded; Max num-threads is 160)
```

**Comment by martin-frbg** [comment]
Great, thanks again.

**Pull request merged**